File: androidApp/src/main/java/com/teewhydope/app/ui/character/mapper/PeoplePresentationToUiModelMapper.kt
package com.teewhydope.app.ui.character.mapper

import com.teewhydope.app.common.mapper.PresentationToUiMapper
import com.teewhydope.app.presentation.character.viewmodel.PeopleListViewState
import com.teewhydope.app.presentation.character.viewmodel.PersonDetailViewState
import com.teewhydope.app.presentation.character.viewmodel.PersonDetailViewState.Loaded
import com.teewhydope.app.presentation.common.mapper.DomainToPresentationMapper
import com.teewhydope.app.ui.character.model.PeopleUiModel


class PeoplePresentationToUiModelMapper(
    private val personDetailPresentationToUiModelMapper: PersonDetailPresentationToUiModelMapper
) : PresentationToUiMapper<PeopleListViewState.Loaded, PeopleUiModel>() {
    override fun map(input: PeopleListViewState.Loaded) = PeopleUiModel(
        count = input.people.count,
        results = input.people.results.map { result ->
            personDetailPresentationToUiModelMapper.toUi(Loaded(result))
        }
    )
}

File: iosApp/iosApp/People/Mapper/PeoplePresentationToUiModelMapper.swift
//
//  PeoplePresentationToUiModelMapper.swift
//  iosApp
//
//  Created by ibrahim animasahun on 09/04/2025.
//  Copyright © 2025 orgName. All rights reserved.
//


import Foundation
import CommonData

class PeoplePresentationToUiModelMapper: PresentationToUiMapper{
    typealias T = PeopleListViewStateLoaded
    
    typealias U = PeopleUiModel
    
    private let personDetailPresentationToUiModelMapper: PersonDetailPresentationToUiModelMapper
    
    init(
        personDetailPresentationToUiModelMapper: PersonDetailPresentationToUiModelMapper,
    ) {
        self
            .personDetailPresentationToUiModelMapper = personDetailPresentationToUiModelMapper
    }
    
    func toUi(input: PeopleListViewStateLoaded) -> PeopleUiModel {
        
        return .init(
            count: Int(input.people.count),
            results: input.people.results.map {
                personDetailPresentationToUiModelMapper
                    .toUi(input: PersonDetailViewStateLoaded(personDetail: $0) )
            }
        )
    }
    
}

